src/source/Dtls/Dtls.c

```diff
@@ -342,13 +362,21 @@ STATUS dtlsSessionProcessPacket(PDtlsSession pDtlsSession, PBYTE pData, PINT32 p
         LOG_OPENSSL_ERROR("SSL_read");
     }
 
-    *pDataLen = sslRet;
-    CHK_STATUS(dtlsCheckOutgoingDataBuffer(pDtlsSession));
+    if (!SSL_is_init_finished(pDtlsSession->pSsl)) {
+        CHK_STATUS(dtlsCheckOutgoingDataBuffer(pDtlsSession));
+    } else {
+        // if dtls handshake is done, and SSL_read did not fail, then sslRet and number of sctp bytes read
+        dataLen = sslRet < 0 ? 0 : sslRet;
+    }
 
 CleanUp:
 
     CHK_LOG_ERR(retStatus);
 
+    if (pDataLen != NULL) {
```

Comment:
I would drop `pDataLen`, we have a `CHK` call when we enter the function

Reply:
fixed